allennlp/training/trainer.py

```diff
@@ -458,14 +458,14 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
                 self._optimizer.step()
 
             # Update the description with the latest metrics
-            metrics = self._get_metrics(train_loss, batch_num)
```

Comment:
Oh, sorry, looks like I misled you with my question.  There was a good reason that we had both of them.  We need to keep this distinction - `batch_num` is used to divide the loss by the number of batches, and really needs to be reset every epoch, while `batch_num_total` is used to keep a global counter for tensorboard.  The names are so similar, though, that it wasn't obvious that they were both necessary.  Maybe `batches_this_epoch` and `total_batches_so_far`?

Reply:
ugh, i'm dumb.

Reply:
No, you're really good at this.  It's just that names are important, or it's really easy to misunderstand what code is doing - I misunderstood it too, at first.